test/python/quantum_info/operators/symplectic/test_clifford.py

```diff
@@ -181,23 +182,56 @@ def test_append_1_qubit_gate(self):
+            "s": PauliList(["+Y", "+Z"]),
+            "sdg": PauliList(["-Y", "+Z"]),
+            "sinv": PauliList(["-Y", "+Z"]),
+            "v": PauliList(["+Y", "+X"]),
+            "w": PauliList(["+Z", "+Y"]),
+        }
+        target_tableau = {
+            "i": np.array([[True, False], [False, True]], dtype=bool),
+            "id": np.array([[True, False], [False, True]], dtype=bool),
+            "iden": np.array([[True, False], [False, True]], dtype=bool),
+            "x": np.array([[True, False], [False, True]], dtype=bool),
+            "y": np.array([[True, False], [False, True]], dtype=bool),
+            "z": np.array([[True, False], [False, True]], dtype=bool),
+            "h": np.array([[False, True], [True, False]], dtype=bool),
+            "s": np.array([[True, True], [False, True]], dtype=bool),
+            "sdg": np.array([[True, True], [False, True]], dtype=bool),
+            "sinv": np.array([[True, True], [False, True]], dtype=bool),
+            "v": np.array([[True, True], [True, False]], dtype=bool),
+            "w": np.array([[False, True], [True, True]], dtype=bool),
+        }
 
         for gate_name in ("i", "id", "iden", "x", "y", "z", "h", "s", "sdg", "v", "w"):
             with self.subTest(msg="append gate %s" % gate_name):
                 cliff = Clifford([[1, 0], [0, 1]])
                 cliff = _append_circuit(cliff, gate_name, [0])
-                value_table = cliff.table._array
-                value_phase = cliff.table._phase
-                value_stabilizer = cliff.stabilizer.to_labels()
-                value_destabilizer = cliff.destabilizer.to_labels()
-                self.assertTrue(np.all(np.array(value_table == target_table[gate_name])))
-                self.assertTrue(np.all(np.array(value_phase == target_phase[gate_name])))
-                self.assertTrue(
-                    np.all(np.array(value_stabilizer == [target_stabilizer[gate_name]]))
-                )
-                self.assertTrue(
-                    np.all(np.array(value_destabilizer == [target_destabilizer[gate_name]]))
-                )
+                # Deprecated methods
+                with self.assertWarns(DeprecationWarning):
+                    value_table = cliff.table._array
+                    value_phase = cliff.table._phase
+                    value_stabilizer = cliff.stabilizer.to_labels()
+                    value_destabilizer = cliff.destabilizer.to_labels()
+                    self.assertTrue(np.all(np.array(value_table == target_table[gate_name])))
+                    self.assertTrue(np.all(np.array(value_phase == target_phase[gate_name])))
+                    self.assertTrue(
+                        np.all(np.array(value_stabilizer == [target_stabilizer[gate_name]]))
+                    )
+                    self.assertTrue(
+                        np.all(np.array(value_destabilizer == [target_destabilizer[gate_name]]))
+                    )
```

Comment:
We should keep the `with` block localised as small as possible, and also have direct tests that the new, alternative paths for `stabilizers()` and so on also work.